subcommand/connect-init/command_test.go

```diff
@@ -45,140 +45,230 @@ func TestRun_FlagValidation(t *testing.T) {
 	}
 }
 
-// TestRun_ServicePollingWithACLs bootstraps and starts a consul server using a mock
+// TestRun_ServicePollingWithACLsAndTLS bootstraps and starts a consul server using a mock
 // kubernetes server to provide responses for setting up the consul AuthMethod
-// then validates that the command runs end to end successfully.
-func TestRun_ServicePollingWithACLs(t *testing.T) {
+// then validates that the command runs end to end successfully. Also tests with TLS on/off.
+func TestRun_ServicePollingWithACLsAndTLS(t *testing.T) {
 	t.Parallel()
-	bearerFile := common.WriteTempFile(t, serviceAccountJWTToken)
-	proxyFile := common.WriteTempFile(t, "")
-	tokenFile := common.WriteTempFile(t, "")
-
-	// Start Consul server with ACLs enabled and default deny policy.
-	var masterToken = "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586"
-	server, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
-		c.ACL.Enabled = true
-		c.ACL.DefaultPolicy = "deny"
-		c.ACL.Tokens.Master = masterToken
-	})
-	defer server.Stop()
-	require.NoError(t, err)
-	server.WaitForLeader(t)
-	consulClient, err := api.NewClient(&api.Config{Address: server.HTTPAddr, Token: masterToken})
-	require.NoError(t, err)
-
-	// Start the mock k8s server.
-	k8sMockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Set("content-type", "application/json")
-		if r != nil && r.URL.Path == "/apis/authentication.k8s.io/v1/tokenreviews" && r.Method == "POST" {
-			w.Write([]byte(tokenReviewFoundResponse))
-		}
-		if r != nil && r.URL.Path == "/api/v1/namespaces/default/serviceaccounts/counting" && r.Method == "GET" {
-			w.Write([]byte(readServiceAccountFound))
-		}
-	}))
-	defer k8sMockServer.Close()
-
-	// Set up Consul's auth method.
-	authMethodTmpl := api.ACLAuthMethod{
-		Name:        testAuthMethod,
-		Description: "Kubernetes Auth Method",
-		Type:        "kubernetes",
-		Config: map[string]interface{}{
-			"Host":              k8sMockServer.URL,
-			"CACert":            serviceAccountCACert,
-			"ServiceAccountJWT": serviceAccountJWTToken,
+	cases := []struct {
+		name   string
+		secure bool
```

Comment:
we use `secure` in the acceptance tests to mean acls & tls so maybe we can call this `tls` just to avoid confusion.

Reply:
applies below too

Reply:
good point, will change, thanks!